File: searching/js/helpers/ArrayBuilder.js
"use strict";

const orderedSequentialArray = (length = 0) => [...Array(length).keys()];

const orderedRandomArray = (length = 0, max = 100) =>
    unorderedRandomArray(length, max).sort();

const unorderedRandomArray = (length = 0, max = 100) =>
    Array.from({ length: length }, () => Math.floor(Math.random() * max));

module.exports = {
    orderedSequentialArray,
    orderedRandomArray,
    unorderedRandomArray
};

File: searching/js/helpers/array_builder.rb

class ArrayHelper  
  def self.orderedSequentialArray(min: 0, max: 100)
    return [] if min > max
    (min..max).to_a
  end
  
  def self.orderedRandomArray(min: 0, max: 100)
    return [] if min > max
    (min..max).map {|x| rand(min..max)}.sort
  end
  
  def self.unorderedRandomArray(min: 0, max: 100)
    return [] if min > max
    (min..max).map {|x| rand(min..max)}
  end
  
end
